### PR TITLE
Hide sensitive parameters

### PR DESCRIPTION
**Pull request by waipeng**
## Overview

This pull request introduces hiding of Sensitive data output.

Puppet now has type Sensitive, which hides the content from logs, etc. Hide it similarly in output of octocatalog-diff, by using a trivial hash so that any diff will still be show.

See also: https://puppet.com/docs/puppet/5.3/lang_data_sensitive.html
## Checklist

- [x] Make sure that all of the tests pass, and fix any that don't. Just run `rake` in your checkout directory, or review the CI job triggered whenever you push to a pull request.
- [x] Make sure that there is 100% [test coverage](https://github.com/github/octocatalog-diff/blob/master/doc/dev/coverage.md) by running `rake coverage:spec` or ignoring untestable sections of code with `# :nocov` comments. If you need help getting to 100% coverage please ask; however, don't just submit code with no tests.
- [x] If you have added a new command line option, we would greatly appreciate a corresponding [integration test](https://github.com/github/octocatalog-diff/blob/master/doc/dev/integration-tests.md) that exercises it from start to finish. This is optional but recommended.
- [x] If you have added any new gem dependencies, make sure those gems are licensed under the MIT or Apache 2.0 license. We cannot add any dependencies on gems licensed under GPL.
- [x] If you have added any new gem dependencies, make sure you've checked in a copy of the `.gem` file into the [vendor/cache](https://github.com/github/octocatalog-diff/tree/master/vendor/cache) directory.

### CODE REVIEW

**Comment by alexjfisher** [comment]
This works well with `Sensitive[String]` parameters, but not so much with `Sensitive[Hash]`...

```
TypeError: no implicit conversion of Hash into String
  /home/fishera/control-repo/vendor/bundle/ruby/2.5.0/bundler/gems/octocatalog-diff-957a8175d326/lib/octocatalog-diff/catalog-diff/differ.rb:470:in `digest'
```

File: lib/octocatalog-diff/catalog-diff/differ.rb [hunk 1]
```diff
@@ -456,10 +457,18 @@ def ignored?(diff)
...
+      # @param sensitive_parameters [Array] Array of sensitive parameters
       # @return [Hash] Cleaned parameters hash (original input hash is not altered)
-      def cleanse_parameters_hash(parameters_hash)
+      def cleanse_parameters_hash(parameters_hash, sensitive_parameters)
         result = parameters_hash.dup
 
+        # hides sensitive params. We still need to know if there's a going to
+        # be a diff, so we hash the value.
+        sensitive_parameters.each do |p|
+          md5 = Digest::MD5.hexdigest result[p]
```
Comment (alexjfisher) [inline lib/octocatalog-diff/catalog-diff/differ.rb]: To cope with Sensitive[Hash] parameters.  How about...
```suggestion
          md5 = Digest::MD5.hexdigest Marshal.dump(result[p])
```
?

**Comment by ahayworth** [comment]
@waipeng Thank you for this! We'll merge it into the 1.6.0 release. I did have to make a change on the release branch to get tests to pass - take a look [here](https://github.com/github/octocatalog-diff/pull/220/commits/02018babe87fdda81d948a8d5838fb448f678ca3). I think that's a benign change, just a copying/pasting error most likely - but please do let me know if it seems wrong.

**Comment by waipeng** [comment]
Oh yes I forgot to update the test after [this change](https://github.com/github/octocatalog-diff/pull/216/commits/68d2e8620054529734efdee95e2f44a9a5cdf165). Thanks for maintaining this software, it is very useful to us!

**Pull request merged**